kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/HasMetadataOperation.java

```diff
@@ -78,8 +78,10 @@ public T replace(T item) {
         if (got == null) {
           return null;
         }
+
+        item = handleReplace(item);
```

Comment:
It's important the resource version is updated before sending the request to increase the chances of successful replace. Agree the original item shouldn't be touched. Can we possibly switch to the doneable approach used in edit above, but updating resource version & calling done in this method, visitor actually calling handleReplace? Hope that makes sense!